system/Commands/Translation/LocalizationFinder.php

```diff
@@ -0,0 +1,330 @@
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\Commands\Translation\LocalizationFinder\ArrayHelper;
+use Config\App;
+use Locale;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use SplFileInfo;
+
+class LocalizationFinder extends BaseCommand
+{
+    protected $group       = 'Translation';
+    protected $name        = 'lang:find';
+    protected $description = 'Find and save available phrases to translate.';
+    protected $usage       = 'lang:find [options]';
+    protected $arguments   = [];
+    protected $options     = [
+        '--locale'   => 'Specify locale (en, ru, etc...) to save files.',
+        '--dir'      => 'Directory for searching for translations (in relation to the APPPATH).',
+        '--show-new' => 'See only new translations as table. Do not write to files.',
+        '--verbose'  => 'Output detailed information.',
+    ];
+
+    /**
+     * Flag for output detailed information
+     */
+    private bool $verbose = false;
+
+    /**
+     * Flag for showing only translations, without saving
+     */
+    private bool $showNew = false;
+
+    private string $languagePath;
+
+    public function run(array $params)
+    {
+        $this->verbose      = array_key_exists('verbose', $params);
+        $this->showNew      = array_key_exists('show-new', $params);
+        $optionLocale       = $params['locale'] ?? null;
+        $optionDir          = $params['dir'] ?? null;
+        $currentLocale      = Locale::getDefault();
+        $currentDir         = APPPATH;
+        $this->languagePath = $currentDir . 'Language';
+
+        if (ENVIRONMENT === 'testing') {
+            $currentDir         = SUPPORTPATH . 'Services/';
+            $this->languagePath = SUPPORTPATH . 'Language';
+        }
+
+        if (is_string($optionLocale)) {
+            if (! in_array($optionLocale, config(App::class)->supportedLocales, true)) {
+                CLI::error(
+                    'Error: "' . $optionLocale . '" is not supported. Supported locales: '
+                    . implode(', ', config(App::class)->supportedLocales)
+                );
+
+                return EXIT_USER_INPUT;
+            }
+
+            $currentLocale = $optionLocale;
+        }
+
+        if (is_string($optionDir)) {
+            $tempCurrentDir = realpath($currentDir . $optionDir);
+
+            if (false === $tempCurrentDir) {
+                CLI::error('Error: Directory must be located in "' . $currentDir . '"');
+
+                return EXIT_USER_INPUT;
+            }
+
+            if ($this->isSubDirectory($tempCurrentDir, $this->languagePath)) {
+                CLI::error('Error: Directory "' . $this->languagePath . '" restricted to scan.');
+
+                return EXIT_USER_INPUT;
+            }
+
+            $currentDir = $tempCurrentDir;
+        }
+
+        $this->process($currentDir, $currentLocale);
+
+        CLI::write('All operations done!');
+
+        return EXIT_SUCCESS;
+    }
+
```

Comment:
Put the `process` method here to ease the flow. Private methods should be at least arranged in the order of use to easily see the flow of logic.